### DISCUSSION TITLE
Combobox with tabs?

### DISCUSSION THREAD

**Discussion by PR7JW7L** [ariakit/ariakit#2270] [Questions]
![image](https://user-images.githubusercontent.com/67312428/232265330-04ea7d69-1835-4e23-a89a-edf51e3b37d7.png)
How do I create a similar pattern found in [https://refero.design/](https://refero.design/) homepage that looks like a combobox with tabs inside?

**Comment by petejodo** [comment]
Working on something similar, in my case though the results per tab is a list of combobox items... I don't suspect ariakit naturally composes well with this (still learning ariakit though). I think a general structure could be like the snippet below but the correct options to use to wire it all together is not so straightforward. The difficult part is keyboard navigation because the input expects left and right arrow keys but may a custom event handler for detecting when the cursor is at the end of the text and the user presses right it removes focus from the input and input the popover and takes the user to the second tab.

```tsx
const combobox = useComboboxStore({ ... });
const tabs = useTabStore({ ... });

return (
  <>
    <Combobox store={combobox} />
    <Popover store={combobox}>
      <TabProvider store={tabs}>
        <TabList>
          <Tab>...</Tab>
          <Tab>...</Tab>
          <Tab>...</Tab>
        </TabList>
        <div>
          <TabPanel>
            <ComboboxList>
              ...items...
            </ComboboxList>
          </TabPanel>
          ...additional tab panels...
        </div>
      </TabProvider>
    </Popover>
  </>
);
```
